### PR TITLE
gRPC proxy

### LINKED ISSUE #5: Create gRPC mapping of DA interface
Define gRPC mapping (protobuf service) of Go DA interface.

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

gRPC proxy allows developers to easily use DA interface over gRPC connection. 
This PR also partially addresses #6 by refactoring test suite. Proxy client/server test shows how easy it is to reuse the test suite.

Resolves #5.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: test/test_suite.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	for i := uint64(1); !found && !time.Now().After(end); i++ {
+		ret, err := da.GetIDs(i)
+		if err != nil {
+			t.Error("failed to get IDs:", err)
+		}
+		if len(ret) > 0 {
+			blobs, err := da.Get(ret)
+			assert.NoError(t, err)
+
+			if len(blobs) == len(msgs) {
```
Comment (tuxcanfly) [inline test/test_suite.go:125]: Would be good have a explanation comment here.

**Review by tuxcanfly — APPROVED** [review]
OK.

File: proto/da/da.proto [hunk 2]
```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+package da;
+
+// DAService is the protobuf service definition for interaction with Data Availability layers.
+service DAService {
+	// Get returns Blob for each given ID, or an error.
```
[the remark this answers is not in the document]
Comment (tzdybal) [reply proto/da/da.proto:6]: Code generated by gRPC always contain errors in method signatures, so we're fine. See: https://github.com/rollkit/go-da/blob/17e0ce611290bda8b4c4172ea3e1e3214804e2d4/types/pb/da/da.pb.go#L733-L747

**Review by Manav-Aggarwal — APPROVED** [review]
LGTM

**Review by nashqueue — APPROVED** [review]
utAck

File: proto/da/da.proto [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	rpc Get(GetRequest) returns (GetResponse) {}
+
+	// GetIDs returns IDs of all Blobs located in DA at given height.
+	rpc GetIDs(GetIDsRequest) returns (GetIDsResponse) {}
+
+	// Commit creates a Commitment for each given Blob.
+	rpc Commit(CommitRequest) returns (CommitResponse) {}
+
+	// Submit submits a Blobs to Data Availability layer.
```
[the remark this answers is not in the document]
Comment (nashqueue) [reply proto/da/da.proto]: follow up https://github.com/rollkit/go-da/issues/15

**Pull request merged**